skfda/misc/operators/_linear_differential_operator.py

```diff
@@ -650,4 +734,21 @@ def custombasis_penalty_matrix_optimized(
     basis: CustomBasis,
 ) -> NDArrayFloat:
     """Optimized version for CustomBasis."""
+    if isinstance(basis.fdata, FDataGrid):
+        product_matrix = np.empty((basis.n_basis, basis.n_basis))
+        operator_evaluated = linear_operator(basis.fdata)(
+            basis.fdata.grid_points[0],
+        )
+
+        # Discard last dimension
+        operator_evaluated = operator_evaluated[..., 0]
+        for i in range(basis.n_basis):
```

Comment:
I think that this can be easily vectorized. In fact, is it not possible to use the function `inner_product_matrix` for this purpose?

Reply:
We could probably discuss this in more detail this week since I do not see exactly how to vectorize it. It is not equivalent to `inner_product_matrix` since when calling `inner_product_matrix`, the grid in which the function is evaluated is not specified. And, in the testing I have performed, that makes a huge difference. Maybe that is because I have tested with very fine grids. However, since the outer loops will have few iterations, especially compared to the integrations across all discretization points, I did not think that the loops were an issue.

Reply:
Can't you construct a `FDataGrid` with that grid and call then `inner_product_matrix`?

Reply:
Maybe that could work. I'll have a look when I have some more time (I have a couple of assignments due this week)

Reply:
Actually yes. That works great. I have changed it in bb059682758b1562a298943ed775fcef551804d3. Thanks!